beacon-chain/slasher/chunks_test.go

```diff
@@ -239,6 +283,49 @@ func TestMaxSpanChunksSlice_CheckSlashable(t *testing.T) {
 	require.NotEqual(t, (*ethpb.AttesterSlashing)(nil), slashing)
 }
 
+func TestMaxSpanChunksSlice_CheckSlashable_DifferentVersions(t *testing.T) {
+	ctx := context.Background()
+	slasherDB := dbtest.SetupSlasherDB(t)
+	params := &Parameters{
+		chunkSize:          4,
+		validatorChunkSize: 2,
+		historyLength:      4,
+	}
+	validatorIdx := primitives.ValidatorIndex(1)
+	source := primitives.Epoch(0)
+	target := primitives.Epoch(3)
+
+	// We create a vote with Phase0 version.
+	att := createAttestationWrapperEmptySig(t, version.Phase0, source, target, nil, nil)
+
+	// We initialize an empty chunks slice and mark an attestation with (source 0, target 3) as attested.
+	chunk := EmptyMinSpanChunksSlice(params)
+	chunkIndex := uint64(0)
+	startEpoch := target
+	currentEpoch := target
+	_, err := chunk.Update(chunkIndex, currentEpoch, validatorIdx, startEpoch, target)
+	require.NoError(t, err)
+
+	// We create a surrounded vote with Electra version.
+	source = primitives.Epoch(1)
+	target = primitives.Epoch(2)
+	surroundedVote := createAttestationWrapperEmptySig(t, version.Electra, source, target, nil, nil)
+
+	// We save the old attestation record, then check if the sorrounded vote is indeed slashable.
+	attData := att.IndexedAttestation.GetData()
+	attRecord := createAttestationWrapperEmptySig(t, version.Phase0, attData.Source.Epoch, attData.Target.Epoch, []uint64{uint64(validatorIdx)}, []byte{1})
+	err = slasherDB.SaveAttestationRecordsForValidators(
+		ctx,
+		[]*slashertypes.IndexedAttestationWrapper{attRecord},
+	)
+	require.NoError(t, err)
+
+	slashing, err := chunk.CheckSlashable(ctx, slasherDB, validatorIdx, surroundedVote)
+	require.NoError(t, err)
+	// The old record should be converted to Electra and the resulting slashing should be an Electra slashing.
+	require.NotEqual(t, (*ethpb.AttesterSlashingElectra)(nil), slashing)
```

Comment:
I think this assert is not valid any more:
If `slashing` is `nil` but with  `(*ethpb.AttesterSlashing)` type, then this `require.NotEqual` will be successful.

(And it seems it is actually the case with this test.)

Reply:
You're right. I decided to use the `reflect` package because I think it's the easiest way to check for a non-nil interface. Simply checking against nil doesn't work

Reply:
Did this test change?

I still see `require.NotEqual(t, (*ethpb.AttesterSlashingElectra)(nil), slashing)` and `slashing` being actually `nil` of type `AttSlashing` in the test.